### ISSUE TITLE
prefetch_related() that uses a Prefetch with a queryset is not joining correctly

### ISSUE THREAD

**Issue by brki** [swisscom/cleanerversion#129]
Here is a failing test that can be added to `versions_tests.tests.test_models.PrefetchingHistoricTests`:
```
    def test_foreign_key_prefetch_with_historic_version(self):
        self.modify_objects()
        historic_city = City.objects.as_of(self.time1).get(identity=self.c1.identity)

        # Test with a simple prefetch
        team = Team.objects.as_of(self.time1).filter(identity=self.t1.identity).prefetch_related('city')[0]
        self.assertIsNotNone(team.city)
        self.assertEquals(team.city.id, historic_city.id)

        # Test with a Prefetch object without a queryset.
        team = Team.objects.as_of(self.time1).filter(
            identity=self.t1.identity
        ).prefetch_related(Prefetch(
            'city',
        ))[0]
        self.assertIsNotNone(team.city)
        self.assertEquals(team.city.id, historic_city.id)

        # Test with a Prefetch object with a queryset.
        team = Team.objects.as_of(self.time1).filter(
            identity=self.t1.identity
        ).prefetch_related(Prefetch(
            'city',
            queryset=City.objects.as_of(self.time1)
        ))[0]
        # This is currently failing:
        self.assertIsNotNone(team.city)
        self.assertEquals(team.city.id, historic_city.id)
```

The second-to-last assertion fails.

Inspecting the generated query, I can see that it tries to select the related objects using ids.  It should however select using identities and an as_of clause.

I suspect that the `VersionedReverseSingleRelatedObjectDescriptor` needs to have a custom `get_prefetch_queryset()` implementation.

It probably makes sense to merge the Django-1.9-compatibility code before fixing this.

**Comment by brki** [comment]
TODO: create a new release.

**Comment by maennel** [comment]
Can be closed with PR #140
